openlibrary/plugins/worksearch/schemes/authors.py

```diff
@@ -51,7 +51,7 @@ def q_to_solr_params(
             ('q', q),
             ('q.op', 'AND'),
             ('defType', 'edismax'),
-            ('qf', 'name alternate_names'),
+            ('qf', 'name alternate_names key'),
```

Comment:
This is a little bit expensive ; since it'll require checking every request against this field. Instead, let's make `get_results` use a regular expression so that if the query is an author ID, it'll perform a specific query (eg `key:"/authors/OL123A"`) instead!